### PR TITLE
meta: use lerna caching

### PR DESCRIPTION
**Pull request by lohart13**
## Pull Request Checklist

- [x] Have you added new tests to prevent regressions?
- [x] If a documentation update is necessary, have you opened a PR to [the documentation repository](https://github.com/sequelize/website/)? 
- [x] Did you update the typescript typings accordingly (if applicable)?
- [ ] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [x] Does the name of your PR follow [our conventions](https://github.com/sequelize/sequelize/blob/main/CONTRIBUTING.md#6-commit-your-modifications)?

## Description Of Change

This PR implements the use of [lerna caching](https://lerna.js.org/docs/concepts/how-caching-works) when running scripts.

## Todos

- [x] Check if this impacts CI jobs

### CODE REVIEW

**Comment by lohart13** [comment]
In the [ci.yaml](https://github.com/sequelize/sequelize/blob/main/.github/workflows/ci.yml#L112-L134) I notice we are only testing typings for `@sequelize/core`.
Does this need to be updated to include `@sequelize/validator.js` as well?
https://github.com/sequelize/sequelize/blob/64a1b82be246f4f6b39012fedebb1ce903128c7d/.github/workflows/ci.yml#L112-L134

**Comment by WikiRik** [comment]
> In the [ci.yaml](https://github.com/sequelize/sequelize/blob/main/.github/workflows/ci.yml#L112-L134) I notice we are only testing typings for `@sequelize/core`.
> Does this need to be updated to include `@sequelize/validator.js` as well?
> https://github.com/sequelize/sequelize/blob/64a1b82be246f4f6b39012fedebb1ce903128c7d/.github/workflows/ci.yml#L112-L134

Yes, we should not scope the typing tests to a single module. Feel free to add it in here

**Comment by lohart13** [comment]
> > In the [ci.yaml](https://github.com/sequelize/sequelize/blob/main/.github/workflows/ci.yml#L112-L134) I notice we are only testing typings for `@sequelize/core`.
> > Does this need to be updated to include `@sequelize/validator.js` as well?
> > https://github.com/sequelize/sequelize/blob/64a1b82be246f4f6b39012fedebb1ce903128c7d/.github/workflows/ci.yml#L112-L134
> 
> Yes, we should not scope the typing tests to a single module. Feel free to add it in here

Ok, I've added this in now

**Review by WikiRik — DISMISSED** [review]
We're not using `yarn test-unit` in CI so that the build step is dependent on it should not be an issue. Might actually be nice to ensure that you are testing the latest code you have written.
And it works great locally

File: nx.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "prod": ["!{projectRoot}/test/**/*"]
+  },
+  "targetDefaults": {
+    "build": {
+      "dependsOn": ["^build"],
+      "inputs": ["prod", "^prod"],
+      "outputs": ["{projectRoot}/lib", "{projectRoot}/types"]
+    },
+    "test-typings": {
+      "inputs": ["prod", "^prod"]
```
Comment (ephys) [inline nx.json:14]: does this mean it's not going to run typing tests if test files have changed?

Comment (lohart13) [reply nx.json:14]: Good point. I've updated this now to be the same as `test-unit`.

The `test-unit` and `test-typings` scripts will now run when any file changes in the package or when a non-test file in a related package changes.

**Review by WikiRik — APPROVED** [review]

**Comment by WikiRik** [comment]
@lohart13 recent CI jobs are failing because the cache from `build` is generated on a different machine than is used for `test-unit`. See https://nx.dev/recipes/troubleshooting/unknown-local-cache and the following error message;
```
lerna notice cli v7.1.5
lerna info ci enabled
lerna notice filter including "@sequelize/validator.js"
lerna info filter [ '@sequelize/validator.js' ]

 >  Lerna (powered by Nx)   Running target test-unit for project @sequelize/validator.js and 2 tasks it depends on:

 

 

 >  Lerna (powered by Nx)   Successfully ran target test-unit for project @sequelize/validator.js and 2 tasks it depends on



 >  Lerna (powered by Nx)   Invalid Cache Directory for Task "@sequelize/core:build"

   The local cache artifact in "/home/runner/work/sequelize/sequelize/node_modules/.cache/nx/2670392751" was not been generated on this machine.
   As a result, the cache's content integrity cannot be confirmed, which may make cache restoration potentially unsafe.
   If your machine ID has changed since the artifact was cached, run "nx reset" to fix this issue.
   Read about the error and how to address it here: https://nx.dev/recipes/troubleshooting/unknown-local-cache
   
   Pass --verbose to see the stacktrace.

Error: Process completed with exit code 1.
```

Maybe we can disable caching when done on CI? Could you take a look at that?

**Pull request merged**